packages/pwa-kit-dev/src/utils/script-utils.ts

```diff
@@ -290,18 +291,45 @@ export const createBundle = async ({
                 } = await getProjectPkg()
                 const extendsTemplate = 'node_modules/' + ccExtensibility.extends
 
+                const pwaKitDeps = [
+                    '@salesforce/pwa-kit-dev',
+                    '@salesforce/pwa-kit-runtime',
+                    '@salesforce/pwa-kit-react-sdk'
+                ]
+
                 let cc_overrides: string[] = []
-                if (ccExtensibility.overridesDir) {
+                const packageVersions: {[key: string]: string} = {}
+
+                if (ccExtensibility.extends) {
+                    const extendsTemplateVersion =
+                        dependencies[ccExtensibility.extends] ||
+                        devDependencies[ccExtensibility.extends]
                     const overrides_files = await walkDir(
                         ccExtensibility.overridesDir,
                         ccExtensibility.overridesDir
                     )
                     cc_overrides = Array.from(overrides_files).filter((item) =>
                         existsSync(path.join(extendsTemplate, item))
                     )
+
+                    // If the project is using template extensbility, we must use 'npm info' to find the versions of pwa-kit packages
+                    try {
+                        const packageInfo = JSON.parse(
+                            execSync(
+                                `npm info ${ccExtensibility.extends}@${extendsTemplateVersion} --json`
+                            ).toString()
+                        )
+                        pwaKitDeps.forEach((dep) => {
+                            if (packageInfo.dependencies?.[dep]) {
+                                packageVersions[dep] = packageInfo.dependencies[dep]
+                            }
+                        })
+                    } catch (error) {
+                        console.error('Error:', error)
+                    }
                 }
                 bundle_metadata = {
-                    dependencies: {...dependencies, ...devDependencies},
+                    dependencies: {...dependencies, ...devDependencies, ...packageVersions},
```

Comment:
```suggestion
                    dependencies: {...packageVersions, ...dependencies, ...devDependencies},
```
If a project declares a PWA Kit dep directly, we should probably report that version, rather than the one we extract from its nested dependencies.